src/features/preferences/renderer/preference-items/terminal/terminal-font-family/terminal-font-family.tsx

```diff
@@ -5,66 +5,31 @@
 import React from "react";
 import { SubTitle } from "../../../../../../renderer/components/layout/sub-title";
 import { withInjectables } from "@ogre-tools/injectable-react";
-import type { UserStore } from "../../../../../../common/user-store";
-import userStoreInjectable from "../../../../../../common/user-store/user-store.injectable";
 import { observer } from "mobx-react";
-import type { SelectOption } from "../../../../../../renderer/components/select";
 import { Select } from "../../../../../../renderer/components/select";
-import type { Logger } from "../../../../../../common/logger";
-import { action } from "mobx";
-import loggerInjectable from "../../../../../../common/logger.injectable";
-import {
-  terminalFontsInjectable,
-} from "../../../../../../renderer/components/dock/terminal/terminal-fonts.injectable";
+import type { TerminalFontPreferenceModel } from "./terminal-font-options.injectable";
+import terminalFontPreferenceModelInjectable from "./terminal-font-options.injectable";
 
 interface Dependencies {
-  userStore: UserStore;
-  logger: Logger;
-  terminalFonts: Map<string, string>;
+  model: TerminalFontPreferenceModel;
 }
 
-const NonInjectedTerminalFontFamily = observer(
-  ({ userStore, logger, terminalFonts }: Dependencies) => {
-    const bundledFonts: SelectOption<string>[] = Array.from(terminalFonts.keys()).map(font => {
-      const { fontFamily, fontSize } = userStore.terminalConfig;
-
-      return {
-        label: (
-          <span style={{ fontFamily: `${font}, var(--font-terminal)`, fontSize }}>
-            {font}
-          </span>
-        ),
-        value: font,
-        isSelected: fontFamily === font,
-      };
-    });
-
-    const onFontFamilyChange = action(({ value: fontFamily }: SelectOption<string>) => {
-      logger.info(`setting terminal font to ${fontFamily}`);
-
-      userStore.terminalConfig.fontFamily = fontFamily; // save to external storage
-    });
-
-
-    return (
-      <section>
-        <SubTitle title="Font family" />
-        <Select
-          themeName="lens"
-          controlShouldRenderValue
-          value={userStore.terminalConfig.fontFamily}
-          options={bundledFonts}
-          onChange={onFontFamilyChange as any}
-        />
-      </section>
-    );
-  },
+const NonInjectedTerminalFontFamily = observer(({ model }: Dependencies) => (
+  <section>
+    <SubTitle title="Font family" />
+    <Select
+      themeName="lens"
+      controlShouldRenderValue
+      value={model.current.get()}
+      options={model.options.get()}
+      onChange={model.set}
+    />
+  </section>
+),
 );
 
 export const TerminalFontFamily = withInjectables<Dependencies>(NonInjectedTerminalFontFamily, {
   getProps: (di) => ({
-    userStore: di.inject(userStoreInjectable),
-    logger: di.inject(loggerInjectable),
-    terminalFonts: di.inject(terminalFontsInjectable),
+    model: di.inject(terminalFontPreferenceModelInjectable),
```

Comment:
Where this model would be useful/utilized besides terminal settings?
Currently nowhere and probably would not need anywhere.. So why to make it complex from the beginning and split to maximum amount of files? No benefits currently, imho.

Reply:
For further reading: https://en.wikipedia.org/wiki/Occam's_razor

Reply:
Yes, it's kinda improvement I agree. 
But maybe done too early before any real some needs/system optimizations requirements, meanwhile making a whole system a bit more complex for nothing.